config/params/config.go

```diff
@@ -201,3 +203,56 @@ func (b *BeaconChainConfig) InitializeForkSchedule() {
 	// Set Bellatrix fork data.
 	b.ForkVersionSchedule[bytesutil.ToBytes4(b.BellatrixForkVersion)] = b.BellatrixForkEpoch
 }
+
+// ForkScheduleEntry is a Version+Epoch tuple for sorted storage in an OrderedForkSchedule
+type ForkScheduleEntry struct {
+	Version [4]byte
+	Epoch   types.Epoch
+}
+
+// OrderedForkSchedule provides a type that can be used to sort the fork schedule and find the Version
+// the chain should be at for a given epoch (via VersionForEpoch).
+type OrderedForkSchedule []ForkScheduleEntry
+
+// Len implements the Len method of sort.Interface
+func (o OrderedForkSchedule) Len() int { return len(o) }
+
+// Swap implements the Swap method of sort.Interface
+func (o OrderedForkSchedule) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
+
+// Less implements the Less method of sort.Interface
+func (o OrderedForkSchedule) Less(i, j int) bool { return o[i].Epoch < o[j].Epoch }
+
+var VersionForEpochNotFound = errors.New("could not find an entry in the ForkVersionSchedule")
+
+// VersionForEpoch finds the Version of the last entry in the last that comes before the given epoch
+func (o OrderedForkSchedule) VersionForEpoch(epoch types.Epoch) ([4]byte, error) {
+	for i := range o {
+		// moving from lowest to highest, only consider epochs that start at or before the requested epoch
+		if epoch < o[i].Epoch {
+			continue
+		}
```

Comment:
Yes because we want the entry with the highest epoch out of all the entries with an epoch <= the target. I think this would be clearer if I iterated through the list in reverse.